source/agora/utils/Test.d

```diff
@@ -146,89 +152,110 @@ unittest
-
-    assert(prev_txs.length == 0 || prev_txs.length == Block.TxsInBlock);
-    const TxCount = block_count * Block.TxsInBlock;
-
-    // in unittests we use the following blockchain layout:
-    //
-    // genesis => 8 outputs
-    // txs[0] => spend gen_tx.outputs[0]
-    // txs[1] => spend gen_tx.outputs[1]...
-    // ..
-    // tx[9] => spend tx[0].outputs[0]
-    // tx[10] => spend tx[1].outputs[0]
-    // ..
-    // tx[17] => spend tx[9].outputs[0]
-    // tx[18] => spend tx[10].outputs[0]
-    // ..
-    // therefore the genesis block and the 1st block are unique here,
-    // as the 1st block spends all the genesis outputs via separate
-    // transactions, and subsequent blocks have transactions which
-    // spend the only outputs in the transaction from the previous block
+    import std.range;
 
-    Transaction[] transactions;
+    // The number of prev_txs must be greater than zero and
+    // less than or equal to Block.TxsInBlock.
+    assert((prev_txs.length > 0) && (prev_txs.length <= Block.TxsInBlock));
+
+    // If the number of prev_txs is less than Block.TxsInBlock,
+    // then the number of all outputs are must be greater than or equal to Block.TxsInBlock.
+    assert(
+        (prev_txs.length == Block.TxsInBlock) ||
+        (
+            (prev_txs.length < Block.TxsInBlock) &&
+            (prev_txs.map!(tx => tx.outputs.length).sum() >= Block.TxsInBlock)
+        )
+    );
 
-    // always use the same amount, for simplicity
-    const Amount AmountPerTx = spend_amount / Block.TxsInBlock;
+    Transaction[] transactions;
 
-    foreach (idx; 0 .. TxCount)
+    if (prev_txs.length == Block.TxsInBlock)
     {
-        Input input;
-        if (prev_txs.length == 0)  // refering to genesis tx's outputs
+        foreach (block_idx; 0 .. block_count)
         {
-            input = Input(hashFull(gen_tx), idx.to!uint);
+            foreach (tx_idx; 0 .. Block.TxsInBlock)
+            {
+                transactions ~= TxBuilder(unknowns)
+                    .attach(
+                            (block_idx == 0) ?
+                                prev_txs[tx_idx] :
+                                transactions
+                                    [
+                                        (block_idx - 1) *
+                                        Block.TxsInBlock + tx_idx
+                                    ]
+                        )
+                    .split(toward)
+                    .amount(share)
+                    .sign(type);
+            }
         }
-        else  // refering to tx's in the previous block
+    }
+    else
+    {
+        auto builders = iota(Block.TxsInBlock).map!(_ => new TxBuilder(unknowns)).array;
+        foreach (block_idx; 0 .. block_count)
         {
-            input = Input(hashFull(prev_txs[idx % Block.TxsInBlock]), 0);
-        }
+            uint[] offsets;
+
+            iota(Block.TxsInBlock).each!(_ => offsets ~= 0);
+
+            int getAvableIndex (uint last)
```

Comment:
```suggestion
            int getAvailableIndex (uint last)
```

I suppose that's what you meant ?

Reply:
Why did you mark this comment as resolved but didn't fix the typo?